### PR TITLE
ipasam: make krbtgt TDO principal canonical

### PR DESCRIPTION
**Pull request by abbra**
For the trusted domain object for remote realm, we have to use krbtgt/REMOTE-FLAT-NAME@OUR-REALM as a canonical name.

Fixes: https://pagure.io/freeipa/issue/9471

### CODE REVIEW

**Comment by abbra** [comment]
For what it is worth, changing to serial mode, documentation generator works faster on Azure CI: 
- Before: AzurePipeline.freeipa (Docs) Failing after 16m 
- After: AzurePipeline.freeipa (Docs) Successful in 9m

File: daemons/ipa-sam/ipa_sam.c [hunk 1]
```diff
@@ -2239,7 +2239,7 @@ static bool handle_cross_realm_princs(struct ipasam_private *ipasam_state,
 				 * POSIX and SMB identities associated with our domain but we associate
 				 * them with the trust domain object itself */
 				failed += !set_krb_princ(ipasam_state, tmp_ctx,
-							 princ_l_tdo, l_tdo_alias,
+							 l_tdo_alias, princ_l_tdo,
```
Comment (f-trivino) [inline daemons/ipa-sam/ipa_sam.c:2242]: should we adjust the comment above: 

				/* Second: <REMOTE FLAT NAME>$@<OUR REALM>, enabled by default

				/* Second: <krbtgt/REMOTE FLAT NAME>$@<OUR REALM>, enabled by default

**Review by f-trivino — COMMENTED** [review]
@abbra , I see in the issue ticket you verified with win-2022, and this will fix "WIN2022-HK9Q.TESTkrbtgtIPA1" issue. 

I Just have a doc wise minor comment: just before your changes we could indicate "**krbtgt**/REMOTE FLAT NAME>$@<OUR REALM"

ref the doc fix, it is very welcome :)

**Comment by abbra** [comment]
@f-trivino thank you. I did realise that it does not make sense to update just the comment -- since I switched alias and canonical name, the same change is needed above. As a result, I changed those alias/principal name format assignments too. As a side effect, the change to last chunk in the original modification is not needed anymore -- since name/alias pair is changed, that change comes automatically.

**Comment by f-trivino** [comment]
> @f-trivino thank you. I did realise that it does not make sense to update just the comment -- since I switched alias and canonical name, the same change is needed above. As a result, I changed those alias/principal name format assignments too. As a side effect, the change to last chunk in the original modification is not needed anymore -- since name/alias pair is changed, that change comes automatically.

LGTM. Adding ack.

**Pull request closed without merging**